playbooks/roles/backups/files/backup.py

```diff
@@ -106,6 +108,59 @@ def upload_to_azure_storage(file_path, bucket, account, key):
     logging.info('Upload successful')
 
 
+class StaleBackups(Exception):
+    pass
+
+
+class NoBackupsFound(Exception):
+    pass
+
+
+def monitor_gcloud_backups(bucket, sentry):
+    """
+    Double check the backups in the Google Cloud Storage Bucket
+
+    Finds the most recent backup file and checks that it is less than
+    25 hours old. This gives us something of a "dead man's switch"
+    to alert us if the previous day's backups failed silently.
+
+    For now, we just directly raise a Sentry error. In the future,
+    when we have a decent monitoring system in place, this will
+    instead simply log the time since the most recent backup as
+    a metric there and we can alert off that separately.
+
+        bucket: The name of a Google Cloud Storage bucket.
+        sentry: The sentry client
+    """
+
+    import boto
+    import gcs_oauth2_boto_plugin
+
+    logging.info('checking backups in Google Cloud Storage bucket '
+                 '"{}"'.format(bucket))
+
+    ALLOWED_WINDOW = 25  # hours
+    sentry.extra_context({'bucket': bucket})
+
+    try:
+        gcloud_uri = boto.storage_uri(bucket, 'gs')
+        keys = gcloud_uri.get_all_keys()
+        if len(keys) < 1:
+            raise NoBackupsFound("There are no backup files in the bucket")
+        keys.sort(key=lambda x: x.last_modified)
```

Comment:
I think there are some edge cases here that aren't being addressed. This sorting assumes that all backup files should have the same prefix (e.g. something like `prod-ou-ficus-mongo-0_mongodb_TIMESTAMP`). I think this assumption will fail for these cases:

1. Basic deployments (a single VM with dbs running locally) will store both Mongo and MySQL in this bucket
2. Enterprise deployments (multiple edxapp/mongo VMs) will store a Mongo backup from each Mongo VM
3. A bucket is reused after an upgrade and contains backups from the old VM(s). The filename is based on the hostname, which often changes between upgrades. See the `make_file_name` function in this file. 

We have a small number of customers who fall under No.1 and only a single Enterprise customer for No.2, but this check will potentially fail for them. 

I think the best approach here might be to construct the backup's filename, minus the timestamp, via `make_file_name`, then first filter the list of backups with that filename.

Reply:
Yeah, I was hoping I could avoid that :) I spot checked a couple and saw no mysql backups in the bucket so I figured they were all switched over to cloud SQL. Reusing buckets with the old backups I think isn't an issue since those will necessarily be older files than the backups that we want. I was unaware of 2 though, so that will definitely need to be addressed.